lib/beaker/dsl/helpers/puppet_helpers.rb

```diff
@@ -606,7 +606,10 @@ def stub_forge_on(machine, forge_host = nil)
           #use global options hash
           primary_forge_name = 'forge.puppetlabs.com'
           forge_host ||= options[:forge_host]
-          @forge_ip ||= Resolv.getaddress(forge_host)
+          ipv4_re = /(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)/
+          match = curl_on(machine, "--verbose #{forge_host}", :accept_all_exit_codes => true).stderr.match(ipv4_re)
+          raise "Failed to resolve forge host '#{forge_host}'" unless match
+          @forge_ip ||= match[0]
```

Comment:
@johnduarte this all looks good, but could we pull these blocks into a separate helper method in this file to be used by both methods since it's duplicate code (especially the gross regex)?